lib/measured/unit_system_builder.rb

```diff
@@ -8,12 +8,36 @@ def unit(unit_name, aliases: [], value: nil)
     nil
   end
 
+  def si_unit(unit_name, aliases: [], value: nil, range: range = 3)
+    @units.concat build_si_units(unit_name, aliases: aliases, value: value, range: range)
+    nil
+  end
+
   def build
     Measured::UnitSystem.new(@units)
   end
 
   private
 
+  def build_si_units(name, aliases: [], value: nil)
+    si_units = []
+    get_positive_prefixes(range).each_with_index do |prefix, index|
+      si_units << build_si_unit(prefix + name, concat_prefixes(aliases, prefix), value * power(index))
+    end
+
+    get_negative_prefixes(range).each_with_index do |prefix, index|
+      si_units << build_si_unit(prefix + name, concat_prefixes(aliases, prefix), value * power(index, -1))
+    end
```

Comment:
As you can tell, these are very similar. What I'd recommend is actually not distinguishing between positive and negative. Something like this:

```ruby
PREFIXES = [
  ...
  ['n', 'nano', -9],
  ...
  ['k', 'kilo', 3],
  ...
]

def build_si_units(name, aliases: [], value: nil)
  value ||= 1
  PREFIXES.map do |short, long, exponent|
    long_names = aliases.map { |alias| "#{alias}#{name}" }
    build_unit("#{short}#{name}", aliases: long_names, value: value * 10**exponent)
  end
end
```

The tricky part is concatenating the names in the right way. Perhaps `si_unit` should have `short:` and `long:` keyword arguments, so that `build_si_units` can concatenate the prefixes together correctly. I wouldn't want to end up with `kmeter`, but maybe we just call that part out in the README 😄